api/v1alpha1/tenantcontrolplane_funcs.go

```diff
@@ -8,25 +8,25 @@ import (
 
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
+	k8stypes "k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	kamajierrors "github.com/clastix/kamaji/internal/errors"
 )
 
-func (in *TenantControlPlane) GetAddress(ctx context.Context, client client.Client) (string, error) {
+func (in *TenantControlPlane) GetAddress(ctx context.Context, client client.Client, namespacedName k8stypes.NamespacedName, specAddress string) (string, error) {
```

Comment:
If we want to go for delegating the caller to specify the right address, I'd say we should use a function that is not bounded to the TenantControlPlane. Eventually, also an _enum_ could be used.

I'd go for two functions here since the domain is slightly different (CP vs Konnectivity) and eventually we can replicate the same underlying logic by abstracting it.

We just need to avoid passing receiver struct members as parameters.